src/qililab/analog/annealing_program.py

```diff
@@ -57,37 +62,62 @@ def transpile(self, transpiler: Callable):
 
         # iterate over each anneal step and transpile ising to fluxes
         for annealing_step in self._annealing_program:
-            for circuit_element in annealing_step:
+            transpiled_step = {}
+            for chip_element in annealing_step:
                 phix, phiz = transpiler(
-                    delta=annealing_step[circuit_element]["sigma_x"], epsilon=annealing_step[circuit_element]["sigma_z"]
+                    delta=annealing_step[chip_element]["sigma_x"], epsilon=annealing_step[chip_element]["sigma_z"]
                 )
-                annealing_step[circuit_element]["phix"] = phix
-                annealing_step[circuit_element]["phiz"] = phiz
+                transpiled_step[f"phix_{self._chip_element_to_short(chip_element)}"] = phix
+                transpiled_step[f"phiz_{self._chip_element_to_short(chip_element)}"] = phiz
 
-    def get_waveforms(self) -> dict[str, tuple[Bus, ArbitraryWave]]:
+            self._transpiled_program.append(transpiled_step.copy())
+
+    def _chip_element_to_short(self, chip_element: str) -> str:
+        """Parse names from algorithm notation (e.g. qubit_0) to runcard notation (e.g. q0)
```

Comment:
Is this not enforcing a runcard notation? What if the runcard element is defined as "q_0"? If we are enforcing a notation this should be documented and properly explained and circulated in the appropiate channels. Changelog is failing to reflect this.

Reply:
This is already in the changelog for the annealing program pr   [#767](https://github.com/qilimanjaro-tech/qililab/pull/767), but if that's not clear we can perhaps modify it / emphasize it on the docstring. We could also look into other ways to enforce some notation / alternatives to this from the code side. This will most likely be dealt with at https://linear.app/qilimanjaro/issue/QHC-694/objectify-analog-program-execution-workflow

Reply:
We should circulate it more, no need for now for extra ways to enforce, but we already introduced some bug in main because not everyone was aware of the flux topology being in the runcard and failed to serialize.